### PR TITLE
Use structured generator for kubectl autoscale

### PR DESCRIPTION
**Pull request by wackxu**
**What this PR does / why we need it**:

Use structured generator for kubectl autoscale

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
part of  # https://github.com/kubernetes/kubectl/issues/138

**Special notes for your reviewer**:
/assign @mengqiy 

**Release note**:

```release-note
Use structured generator for kubectl autoscale 
```

### CODE REVIEW

File: pkg/kubectl/autoscale.go [hunk 1]
```diff
@@ -18,98 +18,68 @@ package kubectl
...
 	return &scaler, nil
 }
+
+// validate check if the caller has set the right fields.
+func (s HorizontalPodAutoscalerGeneratorV1) validate() error {
+	if len(s.Name) == 0 {
+		return fmt.Errorf("name must be specified")
+	}
+	if s.MaxReplicas <= 0 {
+		return fmt.Errorf("'max' is a required parameter and must be greater than zero")
```
Comment (zjj2wry) [inline pkg/kubectl/autoscale.go]: this different with `validateFlags` func in autoscale.go.

Comment (wackxu) [reply pkg/kubectl/autoscale.go]: @zjj2wry Thanks for your review, Done, PTAL

**Comment by mengqiy** [comment]
I suggest you split this PR into 2 PR.
Merge the refactoring half first. This half LG.
Then the `kubectl create` half.

**Comment by wackxu** [comment]
@mengqiy Done, PTAL

**Comment by mengqiy** [comment]
/lgtm

/assign @fabianofranz 
Can you approve this?

**Comment by fabianofranz** [comment]
Thanks @wackxu!
/approve
/sig cli

**Pull request merged**